Delete all legacy sharding related code

mattlord opened the pull request:
Signed-off-by: Matt Lord <mattalord@gmail.com>

<!--
  Thank you for your contribution to the Vitess project.
  How to contribute: https://vitess.io/docs/contributing/
  Please first make sure there is an open Issue to discuss the feature/fix suggested in this PR.
  If this is a new feature, please mark the Issue as "RFC".
 -->

<!-- if this PR is Work in Progress please create it as a Draft Pull Request -->

## Description

<!-- A few sentences describing the overall goals of the pull request's commits. -->
<!-- If this is a bug fix and you think the fix should be backported, please write so. -->

## Related Issue(s)

<!-- List related issues and pull requests. If this PR fixes an issue, please add it using Fixes #????  -->

## Checklist

-   [ ] "Backport me!" label has been added if this change should be backported
-   [ ] Tests were added or are not required
-   [ ] Documentation was added or is not required

## Deployment Notes

<!-- Notes regarding deployment of the contained body of work. These should note any db migrations, etc. -->

ajm188 commented on go/cmd/vtctldclient/command/shards.go:
```diff
@@ -80,11 +80,11 @@ that shard.`,
-Only use this for an emergency fix or after a finished MoveTables. The MigrateServedFrom
-and MigrateServedType commands set this record appropriately already.
+Only use this for an emergency fix or after a finished MoveTables. The MigrateServedType
+commands set this record appropriately already.
```
this should be singular "command" now

mattlord replied:
Actually, this command is now gone so I'll remove the entire sentence. Thanks!

ajm188 commented on go/test/endtoend/cluster/cluster_util.go:
```diff
@@ -20,11 +20,15 @@ import (
 
+	"vitess.io/vitess/go/json2"
 	replicationdatapb "vitess.io/vitess/go/vt/proto/replicationdata"
+	"vitess.io/vitess/go/vt/proto/topodata"
```
this is a dupe of the (_very_ weirdly) renamed `tabletpb` on line 34/38

do you want to tidy up these imports into the usual sections "while you're here"? (please feel free to say "no"!!)

ajm188 commented on go/test/endtoend/cluster/cluster_util.go:
```diff
@@ -292,3 +298,100 @@ func filterDoubleDashArgs(args []string, version int) (filtered []string) {
 	return filtered
 }
+
+// WriteDbCredentialToTmp writes json format db credentials to tmp directory
```
(i know you're just copy-pasting from that now-deleted file, but —) can you punctuate these function comments? ❤️

ajm188 commented on go/test/endtoend/cluster/cluster_util.go:
```diff
@@ -292,3 +298,100 @@ func filterDoubleDashArgs(args []string, version int) (filtered []string) {
+	assert.True(t, reflect.DeepEqual(currentPartition, expectedPartition))
+}
+
+// GetSrvKeyspace return the Srv Keyspace structure
```
```suggestion
// GetSrvKeyspace returns the Srv Keyspace structure.
```

ajm188 commented on go/test/endtoend/cluster/cluster_util.go:
```diff
@@ -292,3 +298,100 @@ func filterDoubleDashArgs(args []string, version int) (filtered []string) {
+	return fmt.Sprintf(pwdChangeCmd, pwdCol, pwdCol, pwdCol, pwdCol, pwdCol, pwdCol)
+}
+
+// getPasswordField Determines which column is used for user passwords in this MySQL version.
```
```suggestion
// getPasswordField determines which column is used for user passwords in this MySQL version.
```

ajm188 commented on go/test/endtoend/tabletmanager/tablet_health_test.go:
```diff
@@ -266,7 +266,7 @@ func TestHealthCheckDrainedStateDoesNotShutdownQueryService(t *testing.T) {
 	// Change from rdonly to drained and stop replication. (These
-	// actions are similar to the SplitClone vtworker command
+	// actions are similar to the old SplitClone vtworker command
 	// implementation.)  The tablet will stay healthy, and the
```
```suggestion
	// implementation). The tablet will stay healthy, and the
```

(i'm **so** sorry, i cannot help myself)

ajm188 commented on go/vt/binlog/keyrange_filter.go:
```diff
@@ -31,7 +31,7 @@ import (
 // passed into the Streamer: bls.Stream(file, pos, sendTransaction) ->
 // bls.Stream(file, pos, KeyRangeFilterFunc(keyrange, sendTransaction))
-func KeyRangeFilterFunc(keyrange *topodatapb.KeyRange, callback func(*binlogdatapb.BinlogTransaction) error) sendTransactionFunc {
+func KeyRangeFilterFunc(keyrange *topodatapb.KeyRange, callback func(*binlogdatapb.BinlogTransaction) error) sendTransactionFunc { // nolint:revive
```
curious what the linter complain here was

mattlord replied:
```
go/vt/binlog/keyrange_filter.go:34:110: unexported-return: exported func KeyRangeFilterFunc returns unexported type binlog.sendTransactionFunc, which can be annoying to use (revive)
```

ajm188 commented on go/vt/binlog/tables_filter.go:
```diff
@@ -34,7 +34,7 @@ const (
 // passed into the Streamer: bls.Stream(file, pos, sendTransaction) ->
 // bls.Stream(file, pos, TablesFilterFunc(sendTransaction))
-func TablesFilterFunc(tables []string, callback func(*binlogdatapb.BinlogTransaction) error) sendTransactionFunc {
+func TablesFilterFunc(tables []string, callback func(*binlogdatapb.BinlogTransaction) error) sendTransactionFunc { // nolint:revive
```
same comment about `nolint` here

mattlord replied:
Answered on the first one.

ajm188 commented on go/vt/wrangler/keyspace.go:
```diff
@@ -45,12 +38,6 @@ const (
 	DefaultFilteredReplicationWaitTime = 30 * time.Second
 )
 
-// TODO(b/26388813): Remove these flags once vtctl WaitForDrain is integrated in the vtctl MigrateServed* commands.
```
lollllllll

ajm188 commented on proto/vtctlservice.proto:
```diff
@@ -202,8 +202,6 @@ service Vtctld {
   // SetKeyspaceServedFrom changes the ServedFromMap manually, and is intended
   // only for emergency fixes. This does not rebuild the serving graph.
   //
```
I think you need to delete this as well (and then regenerate)

ajm188 reviewed (commented):
couple minor things, then let's goooooooo!!!! 

(a deeply heartfelt thanks to drudging through this!!)

mattlord commented:
Thank you for the thorough review, @ajm188 ! I know that this one was difficult as it's so many lines changed (mostly removed) that it's easy to miss something so the fine tooth used is very much appreciated. I believe that I addressed all of your comments here: https://github.com/vitessio/vitess/pull/10278/commits/778b7d1b7dc69805e44d779868102fd585356f70

Please let me know if I missed something though.

ajm188 reviewed (approved):
LGTM!

deepthi commented on go/flags/endtoend/vttablet.txt:
```diff
@@ -748,7 +746,7 @@ Usage of vttablet:
 	the name of a registered security policy to use for controlling access to URLs - empty means allow all for anyone (built-in policies: deny-all, read-only)
   --service_map value
-	comma separated list of services to enable (or disable if prefixed with '-') Example: grpc-vtworker
+	comma separated list of services to enable (or disable if prefixed with '-') Example: grpc-vtgateservice
```
can we change this example to be a vttablet appropriate one? like `grpc-queryservice`

deepthi replied:
hmm.. this is coming from servenv/service_map.go. If we change it there, it changes for all binaries.

deepthi commented on go/test/endtoend/cluster/vtctld_process.go:
```diff
@@ -64,9 +64,6 @@ func (vtctld *VtctldProcess) Setup(cell string, extraArgs ...string) (err error)
 		"--service_map", vtctld.ServiceMap,
 		"--backup_storage_implementation", vtctld.BackupStorageImplementation,
 		"--file_backup_storage_root", vtctld.FileBackupStorageRoot,
-		// hard-code these two soon-to-be deprecated drain values.
-		"--wait_for_drain_sleep_rdonly", "1s",
-		"--wait_for_drain_sleep_replica", "1s",
```
are these flags now deleted? or only deprecated?

mattlord replied:
They're now deleted. They were marked as deprecated before but this end2end test comment wasn't updated: inhttps://github.com/vitessio/vitess/blob/23727bfee12607250da769180fab8064b073dad1/go/vt/wrangler/keyspace.go#L48-L52

deepthi commented on go/test/endtoend/tabletmanager/tablet_health_test.go:
```diff
@@ -266,8 +266,8 @@ func TestHealthCheckDrainedStateDoesNotShutdownQueryService(t *testing.T) {
 	// Change from rdonly to drained and stop replication. (These
-	// actions are similar to the SplitClone vtworker command
-	// implementation.)  The tablet will stay healthy, and the
+	// actions are similar to the old SplitClone vtworker command
```
we can remove the sentence that refers to SplitClone altogether.

mattlord commented:
Thanks, @deepthi ! I believe that I addressed all of your comments here: https://github.com/vitessio/vitess/pull/10278/commits/a9ba0cf3f360a9f9f1627261ac814fe5edc09491

Please let me know if I missed something though.
